controllers/datadogagent/poddisruptionbudget.go

```diff
@@ -33,37 +35,74 @@ type (
 	pdbBuilder func(dda *datadoghqv1alpha1.DatadogAgent) *policyv1.PodDisruptionBudget
 )
 
+type (
+	pdbV1Beta1Builder func(dda *datadoghqv1alpha1.DatadogAgent) *policyv1beta1.PodDisruptionBudget
+)
+
 func (r *Reconciler) manageClusterAgentPDB(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent) (reconcile.Result, error) {
 	if dda.Spec.ClusterAgent.Replicas != nil && *dda.Spec.ClusterAgent.Replicas > 1 {
 		cleanUpCondition := !isClusterAgentEnabled(dda.Spec.ClusterAgent)
-		return r.managePDB(logger, dda, getClusterAgentPDBName(dda), buildClusterAgentPDB, cleanUpCondition)
+		return r.managePDB(logger, dda, getClusterAgentPDBName(dda), buildClusterAgentPDBV1, buildClusterAgentPDBV1Beta1, cleanUpCondition)
 	}
 	return reconcile.Result{}, nil
 }
 
 func (r *Reconciler) manageClusterChecksRunnerPDB(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent) (reconcile.Result, error) {
 	cleanUpCondition := !needClusterChecksRunner(dda)
-	return r.managePDB(logger, dda, getClusterChecksRunnerPDBName(dda), buildClusterChecksRunnerPDB, cleanUpCondition)
+	return r.managePDB(logger, dda, getClusterChecksRunnerPDBName(dda), buildClusterChecksRunnerPDBV1, buildClusterChecksRunnerPDBV1Beta1, cleanUpCondition)
 }
 
-func (r *Reconciler) managePDB(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, pdbName string, builder pdbBuilder, cleanUp bool) (reconcile.Result, error) {
-	if cleanUp {
-		return r.cleanupPDB(dda, pdbName)
-	}
+func (r *Reconciler) managePDB(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, pdbName string, builder pdbBuilder, builderv1beta1 pdbV1Beta1Builder, cleanUp bool) (reconcile.Result, error) {
+	useV1Beta1 := r.platformInfo.UseV1Beta1PDB()
+	if useV1Beta1 {
```

Comment:
This is a good call-out and I agree with you! I anticipate refactor of `poddisruptionbudget.go` when we implement PDBs in v2 since current implementation takes `v1alpha1` CRD, that refactor will take care of two version of PDB as well.